### PR TITLE
Switch to HeartbeatIntervalTimespan as leading value

### PR DESCRIPTION
**Pull request by markusstuhler**

### CODE REVIEW

File: modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/src/Runtime/LegacyCliOptions.cs [hunk 1]
```diff
@@ -244,7 +242,15 @@ public class LegacyCliOptions : Dictionary<string, string>, IAgentConfigProvider
         /// <summary>
         /// The model of the CLI arguments.
         /// </summary>
-        public LegacyCliModel LegacyCliModel { get; }
+        public LegacyCliModel LegacyCliModel {
+            get {
+                if (_legacyCliModel == null) {
+                    _legacyCliModel = ToLegacyCliModel();
+                }
```
Comment (luiscantero) [inline modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/src/Runtime/LegacyCliOptions.cs:249]: Simplify like this?
_legacyCliModel ??= ToLegacyCliModel();

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Storage/PublishedNodesJobConverter.cs [hunk 2]
```diff
@@ -321,18 +320,23 @@ public class OpcNodeModel {
 
             /// <summary> Heartbeat </summary>
             [DataMember(EmitDefaultValue = false)]
-            public int? HeartbeatInterval { get; set; }
+            public int? HeartbeatInterval {
+                get => HeartbeatIntervalTimespan.HasValue ? (int)HeartbeatIntervalTimespan.Value.TotalSeconds : default(int?);
+                set => HeartbeatIntervalTimespan = value.HasValue ? TimeSpan.FromSeconds(value.Value) : default(TimeSpan?);
```
Comment (luiscantero) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Storage/PublishedNodesJobConverter.cs:325]: Simplify with just: default

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Storage/PublishedNodesJobConverter.cs [hunk 3]
```diff
@@ -321,18 +320,23 @@ public class OpcNodeModel {
...
+            }
+
+            /// <summary> Heartbeat </summary>
+            [DataMember(EmitDefaultValue = false)]
+            public int? HeartbeatIntervalMilliSeconds {
+                get => HeartbeatIntervalTimespan.HasValue ? (int)HeartbeatIntervalTimespan.Value.TotalMilliseconds : default(int?);
+                set => HeartbeatIntervalTimespan = value.HasValue ? TimeSpan.FromMilliseconds(value.Value) : default(TimeSpan?);
```
Comment (luiscantero) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Storage/PublishedNodesJobConverter.cs:332]: Simplify with just: default

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Storage/PublishedNodesJobConverter.cs [hunk 4]
```diff
@@ -321,18 +320,23 @@ public class OpcNodeModel {
...
+            }
 
             /// <summary>
             /// Heartbeat interval as TimeSpan.
             /// </summary>
             [IgnoreDataMember]
-            public TimeSpan? HeartbeatIntervalTimespan {
-                get => HeartbeatInterval.HasValue ?
-                    TimeSpan.FromSeconds(HeartbeatInterval.Value) : (TimeSpan?)null;
```
Comment (luiscantero) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Storage/PublishedNodesJobConverter.cs:332]: Simplify with just: default

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Storage/PublishedNodesJobConverter.cs [hunk 5]
```diff
@@ -321,18 +320,23 @@ public class OpcNodeModel {
...
-                set => HeartbeatInterval = value != null ?
-                    (int)value.Value.TotalSeconds : (int?)null;
```
Comment (luiscantero) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Storage/PublishedNodesJobConverter.cs:334]: Simplify with just: default

**Review by luiscantero — CHANGES_REQUESTED** [review]

File: modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/src/Runtime/LegacyCliOptions.cs (hunk 1 above)
Comment (markusstuhler) [reply modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/src/Runtime/LegacyCliOptions.cs:249]: I would say it's preference. ;-)

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Storage/PublishedNodesJobConverter.cs (hunk 2 above)
Comment (markusstuhler) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Storage/PublishedNodesJobConverter.cs:325]: Preference. ;-)

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Storage/PublishedNodesJobConverter.cs (hunk 3 above)
Comment (markusstuhler) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Storage/PublishedNodesJobConverter.cs:332]: see above

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Storage/PublishedNodesJobConverter.cs (hunk 4 above)
Comment (markusstuhler) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Storage/PublishedNodesJobConverter.cs:332]: See above

**Review by luiscantero — DISMISSED** [review]

File: modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/src/Runtime/LegacyCliOptions.cs (hunk 1 above)
Comment (luiscantero) [reply modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/src/Runtime/LegacyCliOptions.cs:249]: Agree ;)

**Review by karok2m — APPROVED** [review]

**Review by luiscantero — APPROVED** [review]

**Pull request merged**